sw-emulator/lib/periph/src/uart.rs

```diff
@@ -12,13 +12,24 @@ Abstract:
 
 --*/
 
-use caliptra_emu_bus::{Bus, BusError};
+use caliptra_emu_bus::{Bus, BusError, Clock, Timer};
+use caliptra_emu_cpu::Irq;
 use caliptra_emu_types::{RvAddr, RvData, RvSize};
+use std::cell::{Cell, RefCell};
+use std::io::Write;
+use std::rc::Rc;
+use std::sync::{Arc, Mutex};
 
 pub struct Uart {
     bit_rate: u8,
     data_bits: u8,
     stop_bits: u8,
+    output: Option<Rc<RefCell<Vec<u8>>>>,
```

Comment:
We don't need these input and output changes in Caliptra I think -- we'll have a separate UART peripheral in MCU.

Caliptra doesn't need input in its UART.

Reply:
Reverted UART back to original.